pkg/client/cache/controller.go

```diff
@@ -25,6 +25,8 @@ import (
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+const stopMarker ExplicitKey = "stopMe"
```

Comment:
Markers like this always bother me . Why can't I plumb through a stop channel to the `Queue` and have it return a typed error for me?